crates/evm/core/src/utils.rs

```diff
@@ -142,7 +143,13 @@ pub fn create2_handler_register<DB: revm::Database, I: InspectorExt<DB>>(
                 .push((ctx.evm.journaled_state.depth(), call_inputs.clone()));
 
             // Sanity check that CREATE2 deployer exists.
-            let code_hash = ctx.evm.load_account(DEFAULT_CREATE2_DEPLOYER)?.0.info.code_hash;
+            // We check which deployer we are using to separate the logic for zkSync and original
+            // foundry.
+            let code_hash = if call_inputs.target_address == DEFAULT_CREATE2_DEPLOYER {
+                ctx.evm.load_account(DEFAULT_CREATE2_DEPLOYER)?.0.info.code_hash
+            } else {
+                ctx.evm.load_account(call_inputs.target_address)?.0.info.code_hash
+            };
```

Comment:
considering that we check if it's the same first, this will be optimized to just do the same as the else branch, making this redundant (also motivation in the other comment)

Reply:
Is this target guaranteed to be `DEFAULT_CREATE2_DEPLOYER_ZKSYNC`? cause then we can use it directly to avoid confusion.

Reply:
Yes, at this point if we are on zksync yes.

Reply:
At this point we would already have change the target_address according to use_zk